internal/controller/instaslice_daemonset.go

```diff
@@ -423,9 +423,15 @@ func calculateTotalMemoryGB(gpuInfoList map[string]string) int {
 func (r *InstaSliceDaemonsetReconciler) discoverMigEnabledGpuWithSlices() ([]string, error) {
 	log := logr.FromContext(context.TODO())
 	instaslice, _, gpuModelMap, failed, errorDiscoveringProfiles := r.discoverAvailableProfilesOnGpus()
-	if failed {
+	if failed || errorDiscoveringProfiles != nil {
```

Comment:
Going by convention, would it make sense to just use `err` ?

Reply:
Ack, I would address it in my next PR